src/coreclr/interpreter/compiler.cpp

```diff
@@ -4911,13 +4937,18 @@ void InterpCompiler::GenerateCode(CORINFO_METHOD_INFO* methodInfo)
                 }
                 m_ip++;
                 break;
+            // NOTE for the below cases: The spec does not say what it means to do a '.un' conversion with a FP operand.
+            // The JIT currently clears the unsigned flag when encountering a fp operand, so the below implementation
+            //  does the equivalent by manually duplicating the conversions from the non-.un opcodes.
```

Comment:
We codified this case a few years ago in the augments: https://github.com/dotnet/runtime/blob/main/docs/design/specs/Ecma-335-Augments.md#unsigned-data-conversion-with-overflow-detection.